### PR TITLE
pkg/securelaunch: integration tests

### PR DESCRIPTION
**Pull request by pjcolp**

### CODE REVIEW

**Comment by pjcolp** [comment]
Hopefully I've written these tests correctly.

**Comment by rminnich** [comment]
They look fine to me. This looks very interesting, I wonder if you could put a bit more context in the comment? I would personally be very interested in the context. Up to you, I'm approving it :-)

File: pkg/securelaunch/policy/policy_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        "type": "file",
+        "location": "/evtlog"
+    }
+}
+`
+
+func TestParse(t *testing.T) {
+	policy, err := parse([]byte(policy_str))
+	if err != nil {
+		t.Fatalf("parse() = %q, not nil", err)
```
Comment (rminnich) [inline pkg/securelaunch/policy/policy_test.go:72]: The %q is a bit surprising, usually it is %v?

**Review by rminnich — COMMENTED** [review]
very interesting!

If you could talk to me a bit about what you're using this for, I'd be interested to hear. I have not kept up with securelaunch status.

File: pkg/securelaunch/policy/policy_test.go (hunk 1 above)
Comment (pjcolp) [reply pkg/securelaunch/policy/policy_test.go]: Ah, yeah. I have %q all over the place. Must've just been a relic from copy/paste. I've updated them to %v

**Comment by pjcolp** [comment]
We've been using this with the TrenchBoot project (https://trenchboot.org/). We use u-root as the initramfs for our SecureLaunch kernel (MLE in Intel parlance) to provide a security engine. Based on a policy file, we can measure various components and ultimately launch (kexec) into a specific target kernel. We can verify that the target kernel/initramfs haven't been tampered with by including hashes in the policy file. We validate the policy file with a signature and validate the public key with a hash stored in the TPM. The stored hash can only be read if the specified PCRs have the correct value (in the default case, we use 17,18,19, which are only available in locality 4 while running the MLE code)

**Review by rminnich — APPROVED** [review]

**Pull request merged**